cylc/flow/network/subscriber.py

```diff
@@ -49,45 +52,45 @@ class WorkflowSubscriber(ZMQSocketBase):
     NOTE: Security to be provided by zmq.auth
 
     Args:
-        host (str):
-            The host to connect to.
-        port (int):
-            The port on the aforementioned host to connect to.
+        host: The host to connect to.
+        port: The port on the aforementioned host to connect to.
 
     Usage:
         * Subscribe to Publisher socket using ``WorkflowSubscriber.__call__``.
 
     """
+    # socket & event loop not None - get assigned on init by self.start():
+    socket: 'zmq.asyncio.Socket'
+    loop: asyncio.AbstractEventLoop
 
     def __init__(
-            self,
-            workflow: str,
-            host: str = None,
-            port: Union[int, str] = None,
-            context: object = None,
-            srv_public_key_loc: str = None,
-            topics: Iterable[bytes] = None
```

Comment:
I never really liked the double indentation .. But the linter use to pull us up otherwise.